packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -866,6 +898,33 @@ class ThemeData with Diagnosticable {
   /// text geometry.
   factory ThemeData.fallback({bool? useMaterial3}) => ThemeData.light(useMaterial3: useMaterial3);
 
+  /// Applies the adaptation if there is one contained in the [adaptationMap].
+  ///
+  /// For example, calling `Theme.of(context).adaptive(switchThemeData)` returns
+  /// an adaptive [SwitchThemeData].
+  T adaptive<T>(T defaultValue) {
```

Comment:
Since different component `.adaptive` constructors might support different platforms, I think it would be better to just provide an adaptationMap lookup like:

```dart
Adaptation<T>? getAdaptation<T>() => adaptationMap[T] as Adaptation<T>?;
```
For Switch.adaptive, if getAdaptation<SwitchThemeData> returned null then we'd use _SwitchThemeAdaptation, which would look like this:

```dart
class _SwitchThemeAdaptation extends Adaptation<SwitchThemeData> {
  const _SwitchThemeAdaptation();

  @override
  SwitchThemeData adapt(ThemeData theme, SwitchThemeData defaultValue) {
    switch (theme.platform) {
      case TargetPlatform.android:
      case TargetPlatform.fuchsia:
      case TargetPlatform.linux:
      case TargetPlatform.windows:
        return defaultValue;
      case TargetPlatform.iOS:
      case TargetPlatform.macOS:
        return const SwitchThemeData();
    }
}
```

Reply:
Thanks a lot for the suggestions! Updated!